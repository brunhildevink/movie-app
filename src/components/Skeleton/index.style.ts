import styled, { keyframes } from 'styled-components'

const skeletonKeyframes = keyframes`
  to {
    background-position-x: -200%;
  }
`

const Wrapper = styled.div<{ height: string; width: string; margin: string }>`
  height: ${({ height }) => height};
  width: ${({ width }) => width};
  margin: ${({ margin }) => margin};
  background: #eee;
  background: linear-gradient(110deg, #ececec 8%, #f5f5f5 18%, #ececec 33%);
  border-radius: 5px;
  background-size: 200% 100%;
  animation: 1.5s ${skeletonKeyframes} linear infinite;
`

export { Wrapper }
