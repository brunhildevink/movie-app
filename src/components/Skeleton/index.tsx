import React from 'react'
import { Wrapper } from './index.style'

interface Props {
  height: string
  width: string
  margin: string
}

const Skeleton: React.FC<Props> = ({ height, width, margin }) => (
  <Wrapper height={height} width={width} margin={margin} />
)

export default Skeleton
