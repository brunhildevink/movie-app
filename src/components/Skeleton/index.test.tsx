import React from 'react'
import renderer from 'react-test-renderer'
import Skeleton from '.'

test('Skeleton renders correctly', () => {
  const tree = renderer.create(<Skeleton height="100px" width="100px" margin="10px" />).toJSON()
  expect(tree).toMatchSnapshot()
})
