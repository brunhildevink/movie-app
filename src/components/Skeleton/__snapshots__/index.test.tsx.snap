// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton renders correctly 1`] = `
<div
  className="sc-bczRLJ iKuZnI"
  height="100px"
  width="100px"
/>
`;
